system/core.go

```diff
@@ -130,3 +135,54 @@ func (application *Application) Route(controller interface{}, route string) inte
 	}
 	return fn
 }
+
+func saveSession(c web.C, w http.ResponseWriter, r *http.Request) error {
+	if session, exists := c.Env["Session"]; exists {
+		return session.(*sessions.Session).Save(r, w)
+	}
+	return log.Errorf("Session not available")
+}
+
+func (application *Application) APIHandler(apiFun func(web.C, *http.Request) *APIResponse) web.HandlerFunc {
+	return func(c web.C, w http.ResponseWriter, r *http.Request) {
+		if err := saveSession(c, w, r); err != nil {
+			log.Errorf("Can't save session: %v", err)
+		}
+
```

Comment:
I was either wrong or had some junk in my tree. Moving saveSession() to after apiFun() does works ok.